src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/GeneralApplicationDirectionsService.java

```diff
@@ -0,0 +1,193 @@
+                                                                  List<BulkPrintDocument> documents) {
+        bulkPrintService.printApplicantDocuments(caseDetails, authorisationToken, documents);
+        bulkPrintService.printRespondentDocuments(caseDetails, authorisationToken, documents);
+    }
+
+    private void resetStateToGeneralApplicationPrestate(CaseDetails caseDetails) {
+        Map<String, Object> caseData = caseDetails.getData();
+        String generalApplicationPreState = (String) caseData.get(GENERAL_APPLICATION_PRE_STATE);
+        if (generalApplicationPreState != null) {
+            caseData.put(STATE, generalApplicationPreState);
+        }
+    }
+
+    private CaseDocument prepareGeneralApplicationDirectionsOrderDocument(CaseDetails caseDetails, String authorisationToken) {
+        CaseDetails caseDetailsCopy = documentHelper.deepCopy(caseDetails, CaseDetails.class);
+        Map<String, Object> caseData = caseDetailsCopy.getData();
+
+        addContestedCourtDetails(caseDetailsCopy);
+        caseData.put("applicantName", DocumentHelper.getApplicantFullName(caseDetailsCopy));
+        caseData.put("respondentName", DocumentHelper.getRespondentFullNameContested(caseDetailsCopy));
+        caseData.put("letterDate", String.valueOf(LocalDate.now()));
+
+        return genericDocumentService.generateDocument(authorisationToken, caseDetailsCopy,
+                documentConfiguration.getGeneralApplicationOrderTemplate(),
+                documentConfiguration.getGeneralApplicationOrderFileName());
+    }
+
+    private CaseDocument prepareHearingRequiredNoticeDocument(CaseDetails caseDetails, String authorisationToken) {
+        CaseDetails caseDetailsCopy = documentHelper.deepCopy(caseDetails, CaseDetails.class);
+        Map<String, Object> caseData = caseDetailsCopy.getData();
+
+        caseData.put("ccdCaseNumber", caseDetails.getId());
+        addContestedCourtDetails(caseDetailsCopy);
+        caseData.put("applicantName", DocumentHelper.getApplicantFullName(caseDetailsCopy));
+        caseData.put("respondentName", DocumentHelper.getRespondentFullNameContested(caseDetailsCopy));
+        addHearingVenueDetails(caseDetailsCopy);
+        caseData.put("letterDate", String.valueOf(LocalDate.now()));
+
+        return genericDocumentService.generateDocument(authorisationToken, caseDetailsCopy,
+                documentConfiguration.getGeneralApplicationHearingNoticeTemplate(),
+                documentConfiguration.getGeneralApplicationHearingNoticeFileName());
+    }
+
+    private void addHearingVenueDetails(CaseDetails caseDetails) {
+        Map<String, Object> caseData = caseDetails.getData();
+        try {
+            Map<String, Object> courtDetailsMap = objectMapper.readValue(getCourtDetailsString(), HashMap.class);
+            Map<String, Object> courtDetails = (Map<String, Object>) courtDetailsMap.get(caseData.get(getSelectedCourt(
+                caseData, "generalApplicationDirections_")));
```

Comment:
don't use hardcoded strings. either create a const or create a method that does this call for you